test/pooler_tests.erl

```diff
@@ -1060,6 +1060,51 @@ pooler_auto_grow_enabled_test_() ->
        end}
      ]}}.
 
+pooler_custom_stop_mfa_test_() ->
+    {foreach,
+     fun() ->
+             Pool = [{name, test_pool_1},
+                     {max_count, 3},
+                     {init_count, 2},
+                     {cull_interval, {200, ms}},
+                     {max_age, {0, min}},
+                     {start_mfa, {pooled_gs, start_link, [{foo_type}]}}],
+             application:set_env(pooler, pools, [Pool])
+     end,
+     fun(_) ->
+             application:unset_env(pooler, pools)
+     end,
+     [
+      {"default behavior kills the pool member",
+       fun() ->
+               application:start(pooler),
```

Comment:
probably worth matching or asserting on the start/stop to make sure we are testing the state we think we are testing.